### ISSUE TITLE
Make labelme work with Polygon-RNN++

### ISSUE THREAD

**Issue by serycjon** [wkentaro/labelme#125]
Let's interconnect these two.
https://github.com/davidjesusacu/polyrnn-pp
https://youtu.be/evGqMnL4P3E

The question is, how to make this optional, so that not everybody has to install tensorflow. Also didn't try polyrnn++ yet, but depending on it computational costs, maybe we should somehow allow for remote execution of the neural network. Any ideas how to do this well?

**Comment by jbvo** [comment]
An idea in the same category that I have been thinking about, is to integrate something like [grabcut](https://en.wikipedia.org/wiki/GrabCut). There's an open source implementation in opencv which works reasonably well, but I haven't really about the technicalities of integrated it yet.

**Comment by serycjon** [comment]
I have a convertor from binary mask to a polygon implemented (using `skimage.measure.find_countours` and `skimage.measure.approximate_polygon`), but I think using grabcut is completely different story compared to polygon-rnn++ (in principle and in expected results quality).

**Comment by wkentaro** [comment]
I added the rectangle annotation proposed in https://github.com/wkentaro/labelme/issues/123.
So it seems we are ready to add polygon rnn++ to labelme.

**Comment by wkentaro** [comment]
I added an experimental version of labelme supporting polygon rnn++: https://github.com/wkentaro/labelme/pull/206#issue-204437736

**Comment by carsonzhu** [comment]
Could you please share the function that converts masks to polygons? @serycjon 
I think this is really a nice function, thanks.

**Comment by serycjon** [comment]
Here you are:
```
def mask_to_polygons(mask, tolerance=2):
    '''Extracts polygons from binary mask

    The mask is a numpy [HxWx1] array with values > 0 meaning mask is
    present.
    '''

    contours = skimage.measure.find_contours(mask, 0.5)
    simplified = []
    for contour in contours:
        new = skimage.measure.approximate_polygon(contour, tolerance)
        simplified.append(new)

    return simplified
```
To get python list with (x, y) coordinates, you than do:
```py_polygons = [poly[:, ::-1].tolist() for poly in polygons]```

**Comment by wkentaro** [comment]
It seems polygon rnn++ is not so useful because of its accuracy for general objects at this point.
I decided not to include it in labelme, so close this.

**Comment by croshawj** [comment]
Is it possible to add the option to create rectangle or create polygon?
